datalake-dr-connector/src/main/java/com/sequenceiq/cloudbreak/datalakedr/converter/GrpcStatusResponseToDatalakeDrStatusResponseConverter.java

```diff
@@ -0,0 +1,24 @@
+package com.sequenceiq.cloudbreak.datalakedr.converter;
+
+import com.cloudera.thunderhead.service.datalakedr.datalakeDRProto;
+import com.sequenceiq.cloudbreak.datalakedr.model.DatalakeDrStatusResponse;
+import java.util.Optional;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GrpcStatusResponseToDatalakeDrStatusResponseConverter {
+
+    public DatalakeDrStatusResponse convert(datalakeDRProto.BackupDatalakeStatusResponse response) {
+        return new DatalakeDrStatusResponse(
+            DatalakeDrStatusResponse.State.valueOf(response.getOverallState()),
+            Optional.ofNullable(response.getFailureReason())
+        );
+    }
+
+    public DatalakeDrStatusResponse convert(datalakeDRProto.RestoreDatalakeStatusResponse response) {
+        return new DatalakeDrStatusResponse(
+            DatalakeDrStatusResponse.State.valueOf(response.getOverallState()),
+            Optional.ofNullable(response.getFailureReason())
```

Comment:
we can not depend on just the getOverallState. Here you need to check the internal states as well.

If the overall status could be failed but the internal states could be still in progress.